dot/core/service_test.go

```diff
@@ -111,7 +112,9 @@ func generateExtrinsic(t *testing.T) (extrinsic, externalExtrinsic types.Extrins
 	require.NoError(t, err)
 
 	encExt := []types.Extrinsic{extEnc.Bytes()}
-	testExternalExt := types.Extrinsic(append([]byte{byte(types.TxnExternal)}, encExt[0]...))
+	testHeader := types.NewEmptyHeader()
+	testExternalExt := types.Extrinsic(append([]byte{byte(types.TxnExternal)},
+		append(encExt[0], testHeader.StateRoot.ToBytes()...)...))
```

Comment:
use a concatenateBytes helper function 😉 
Maybe you can tailor to produce a `types.Extrinsic` since that's all of its usage really.

Same for other uses of `types.Extrinsic(append(` below in this file.